dockertest/subtestbase.py

```diff
@@ -102,10 +161,84 @@ def cleanup(self):
         Always called, before any exceptions thrown are re-raised.
         """
         self.log_step_msg('cleanup')
+        self.overlook_known_failures()
```

Comment:
@edsantiago did you happen to check if there are any sub/sub-subtests which override ``cleanup()`` but don't call ``super()``?

Reply:
Excellent question. No, I hadn't. I've just run `ack -A10 'def cleanup' subtests/` and it looks like there are four:

    login/login.py
    ps_size/ps_size.py
    run_cgroup_parent/run_cgroup_parent.py
    top/top.py

Shall I submit an amendment to this PR? If so, can you remind me of the proper convention for calling `super()` within cleanup? I notice that most call it at the beginning (the usual) but a few call `super()` at the end; and ISTR something special about cleanup. Do you remember what that's about?

Reply:
There may be a reason why they don't call ``super()``, or it was just an omission.  IIRC, in cases where they can't use ``super()`` I'm pretty sure I made an effort to comment about that.  In those cases, they just need to call ``self.overlook_known_failures()`` on their own (assuming ``super()`` isn't applicable).  If they don't override ``cleanup()`` at all, then there's nothing to be done, they'll automatically use the base-class method.  

Yeah, best fix them  in this PR, but if you're in a pinch or any look hairy, let me know and I'll help out.

Reply:
Sorry, I wasn't clear. My question was: do you know why some places call `super()` at the beginning, and some at the end?

    def cleanup(self):
        super(MyBase, self).cleanup()           <---- the normal way

    def cleanup(self):
        ...do some stuff...
        ...do more stuff...
        super(MyBase, self).cleanup()           <---- at the end

Reply:
This could very well be me getting my wires crossed WRT C++.  There, destruction always happens in the reverse (ancestor class method called last) because memory-freeing is involved.  In python, it really only matters if ``...do some stuff...`` is known to negatively affect the state (or behavior) expected by the base-class method.

Here for example, I think it only matters if  if ``...do some stuff...`` involves something that could lead to ``sys.exit(1)`` before ``overlook_known_failures()`` happens.

Otherwise, it's pretty rare to need to not call it not "The normal way", so most likely it was my goof and/or should be commented.  I have a unittest I'm working on elsewhere, that requires having the ``super()`` call in the middle of a ``testcase.setUp()``.  So it does happen.

Reply:
Done.